parse_date.re

```diff
@@ -2231,6 +2378,29 @@ timelib_time *timelib_parse_from_format(char *format, char *string, size_t len,
 		if (s->time->s == TIMELIB_UNSET ) {
 			s->time->s = 0;
 		}
+		if (s->time->us == TIMELIB_UNSET ) {
+			s->time->us = 0;
+		}
```

Comment:
There was originally I believe a reason why I didn't do this for ``us`` (micro seconds), but I didn't leave a comment so ... and I forgot.

Reply:
Yeah I was running into issues on mongo side of things, since we call fromMillisSinceEpoch() with TIMELIB_UNSET (-9999) for the microseconds, which of course is going to return something wonky.